src/QMCWaveFunctions/EinsplineSetBuilderESHDF.fft.cpp

```diff
@@ -157,7 +157,7 @@ bool EinsplineSetBuilder::ReadOrbitalInfo_ESHDF()
       if (Super2Prim[i] < 0)
       {
         app_error() << "Supercell ion " << i << " not found in the primitive cell" << std::endl;
-        abort();
+        // abort();
```

Comment:
checked in by mistake?

Reply:
This is a hack to make the external potential work with no ions.

Reply:
To be more specific, this is to avoid qmcpack from terminating if there are no ions read in from an HDF5 format file that holds SPOs from a Qbox calculation.

Reply:
Then let us introduce an option to allow skipping the ion check.

Reply:
What should the criteria for the skip be? I feel like if no ions are defined in the input file then this error should not be thrown.

Reply:
I got confused. Does your run have no ions from input xml or no ions from h5 generated by qbox?
This check is inside a loop of `for (int i = 0; i < SourcePtcl->R.size(); i++)`. Since you hit the error check, your input xml file has ions. It seems that your h5 has no info about ions in the primitive cell, namely the qbox converter generated h5 is not fully compliant with the h5 generated by QE.
Is "IonPos.size()==0" in your case? IonPos records ion positions loaded from h5.
To workaround the check, you can change `if (Super2Prim.size() == 0` to
```C++
if (IonPos.size()==0)
{
  app_warning() << "Ion positions are missing in h5. Bypassing supercell to primitive cell electron position mapping check!" << std::endl;
}
else if (Super2Prim.size() == 0)
```
In this way, the mapping check is bypassed with a warning printed.

Reply:
there should be no ions in both cases.

Reply:
Then change line 127 to
`if (Super2Prim.size() == 0 && SourcePtcl->R.size() > 0)`

Reply:
I will need to have a look after you fix the test. Keep this change not commited when you are fixing the test.

Reply:
i have changed the test to match the older parameters. I can confirm I get the same answer in both cases with the same parameters.